src/attr.rs

```diff
@@ -18,27 +21,38 @@ impl ToTokens for Display {
 }
 
 pub fn display(attrs: &[Attribute]) -> Result<Option<Display>> {
-    for attr in attrs {
-        if attr.path.is_ident("doc") {
-            let meta = attr.parse_meta()?;
-            let lit = match meta {
-                Meta::NameValue(syn::MetaNameValue {
-                    lit: syn::Lit::Str(lit),
-                    ..
-                }) => lit,
-                _ => unimplemented!(),
-            };
-
-            let lit = LitStr::new(lit.value().trim(), lit.span());
-
+    let attr = attrs
+        .iter()
+        .find(|attr| attr.path.is_ident("display"))
+        .or_else(|| attrs.iter().find(|attr| attr.path.is_ident("doc")));
+    if let Some(attr) = attr {
+        let meta = attr.parse_meta()?;
+        let lit = match meta {
+            Meta::NameValue(MetaNameValue { lit: Str(lit), .. }) => Some(Ok(lit)),
+            Meta::List(MetaList { nested, .. }) => {
+                nested.iter().find_map(|nested_attr| match nested_attr {
+                    NestedMeta::Meta(Meta::Path(path)) => {
+                        if path.is_ident("transparent") {
+                            Some(Ok(LitStr::new("{0}", attr.span())))
```

Comment:
So I think what we might want to do is modify `Display` so instead of only taking a format string, it can take an enum:

```rust
enum DisplayFmt {
    Provided(LitStr),
    Transparent,
}
```

That way we can differentiate between cases where we've been given a format string and cases where we need to figure it out ourselves. Then when we are constructing the actual display implementation for the type if the format is `Transparent` we inspect the struct to look at the members and look for one with a #[source] attribute, or a member named `source`, or if its a unit struct we just insert `"{0}"`.